test/image.bats

```diff
@@ -365,3 +366,149 @@ EOF
 	expected_output=$(date +"%a %b %e %H:%M:%S %Z %Y")
 	[[ "$output" == *"$expected_output"* ]]
 }
+
+@test "run container with memory_limit_in_bytes -1" {
+	cat << EOF > "$CRIO_CONFIG_DIR/99-mem.conf"
+[crio.runtime]
+default_runtime = "mem"
+[crio.runtime.runtimes.mem]
+runtime_path = "$RUNTIME_BINARY_PATH"
+EOF
+	start_crio
+
+	case $(go env GOARCH) in
+	amd64)
+		crictl pull ${IMAGE_LIST_DIGEST_FOR_TAG_AMD64}
+		IMAGE=${IMAGE_LIST_DIGEST_FOR_TAG_AMD64}
+		;;
+	arm64)
+		crictl pull ${IMAGE_LIST_DIGEST_FOR_TAG_ARM64}
+		IMAGE=${IMAGE_LIST_DIGEST_FOR_TAG_ARM64}
+		;;
+	esac
+
+	jq --arg image "$IMAGE" '.metadata.name = "memory"
+		| .command = ["/bin/sh", "-c", "sleep 600"]
+		| .linux.resources.memory_limit_in_bytes = -1
+		| .image.image = $image' \
+		"$TESTDATA"/container_config.json > "$TESTDIR"/memory.json
+
+	run ! crictl run "$TESTDIR"/memory.json "$TESTDATA"/sandbox_config.json
+}
+
+@test "run container with memory_limit_in_bytes 12.5MiB" {
+	cat << EOF > "$CRIO_CONFIG_DIR/99-mem.conf"
+[crio.runtime]
+default_runtime = "mem"
+[crio.runtime.runtimes.mem]
+runtime_path = "$RUNTIME_BINARY_PATH"
+container_min_memory = "7.5MiB"
+EOF
+	start_crio
+
+	case $(go env GOARCH) in
+	amd64)
+		crictl pull ${IMAGE_LIST_DIGEST_FOR_TAG_AMD64}
+		IMAGE=${IMAGE_LIST_DIGEST_FOR_TAG_AMD64}
+		;;
+	arm64)
+		crictl pull ${IMAGE_LIST_DIGEST_FOR_TAG_ARM64}
+		IMAGE=${IMAGE_LIST_DIGEST_FOR_TAG_ARM64}
+		;;
+	esac
+
+	jq --arg image "$IMAGE" '.metadata.name = "memory"
+		| .command = ["/bin/sh", "-c", "sleep 600"]
+		| .linux.resources.memory_limit_in_bytes = 12582912
+		| .image.image = $image' \
+		"$TESTDATA"/container_config.json > "$TESTDIR"/memory.json
+
+	run crictl run "$TESTDIR"/memory.json "$TESTDATA"/sandbox_config.json
+}
+
+@test "run container with container_min_memory 17.5MiB" {
+	cat << EOF > "$CRIO_CONFIG_DIR/99-mem.conf"
+[crio.runtime]
+default_runtime = "mem"
+[crio.runtime.runtimes.mem]
+runtime_path = "$RUNTIME_BINARY_PATH"
+container_min_memory = "17.5MiB"
```

Comment:
This is actually an interesting test.

What should we do if one sets a minimum memory limit higher than what the Pod requests?

I suppose, nothing? But perhaps @haircommander has some thoughts.

Reply:
@saschagrunert, your thoughts?

Reply:
I would do nothing, because it may be intentional.